openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/DataContractRepository.java

```diff
@@ -156,6 +245,405 @@ private Set<String> extractFieldNames(List<org.openmetadata.schema.type.Field> f
     return fieldNames;
   }
 
+  private TestSuite createOrUpdateDataContractTestSuite(DataContract dataContract) {
+    if (nullOrEmpty(dataContract.getQualityExpectations())) {
+      return null; // No quality expectations, no test suite needed
+    }
+    try {
+      String testSuiteName = dataContract.getName() + " - Data Contract Expectations";
+      TestSuiteRepository testSuiteRepository =
+          (TestSuiteRepository) Entity.getEntityRepository(Entity.TEST_SUITE);
+      TestCaseRepository testCaseRepository =
+          (TestCaseRepository) Entity.getEntityRepository(Entity.TEST_CASE);
+
+      // Check if test suite already exists
+      TestSuite testSuite = null;
+      try {
+        testSuite =
```

Comment:
we can write a Optional<T> getEntityorNull in EntityRepository , will make the code more clean , we should catch Exception and return Optional.empty() if